kernel/dev/serial/rs232.cpp

```diff
@@ -82,3 +90,59 @@ void px_rs232_init(uint16_t com_id) {
         "Panix Serial Output Debugger\n\n"
     );
 }
+
+px_ring_buff_t* px_rs232_init_buffer(int size) {
+    // Allocate space for the input buffer
+    read_buffer = (px_ring_buff_t*)malloc(sizeof(px_ring_buff_t));
+    // Initialize the ring buffer
+    if (px_ring_buffer_init(read_buffer, size) == 0) {
+        return read_buffer;
+    } else {
+        return NULL;
+    }
+}
+
+px_ring_buff_t* px_rs232_get_buffer() {
+    // Can return NULL. This is documented.
+    return read_buffer;
+}
+
+char px_rs232_get_char() {
+    // Grab the last byte and convert to a char
+    uint8_t data;
+    px_ring_buffer_dequeue(read_buffer, &data);
+    return (char)data;
```

Comment:
Couldn't this be undefined if the buffer is empty?

Reply:
So how do we know if we received a `null` vs. nothing at all?

Reply:
Added a note in the latest commit. We'll come back to this at a later date when we need better data transmission.